kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/BackoffStrategy.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.time.Duration;
+
+public interface BackoffStrategy {
+
+    /**
+     * Decides how long to delay the next attempt at an action given it has failed
+     * for N consecutive times in the past.
+     * @param failures count of failures
```

Comment:
would you expect failures to be always > 0?

Reply:
I'll document it, it expects `>=0` so you can ask for a backoff on every request without having to special case the first request (0-failures case).